drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -2415,6 +2415,36 @@ int bnxt_vf_default_vnic_count(struct bnxt *bp, uint16_t vf)
 	return count;
 }
 
+/* This function queries the VNIC IDs  for a specified VF. */
+int bnxt_hwrm_func_vf_vnic_query(struct bnxt *bp, uint16_t vf,
+					uint16_t *vnic_ids)
+{
+	struct hwrm_func_vf_vnic_ids_query_input req = {0};
+	struct hwrm_func_vf_vnic_ids_query_output *resp =
+						bp->hwrm_cmd_resp_addr;
+	int rc;
+
+	/* First query all VNIC ids */
+	HWRM_PREP(req, FUNC_VF_VNIC_IDS_QUERY, -1, resp_vf_vnic_ids);
+
+	req.vf_id = rte_cpu_to_le_16(bp->pf.first_vf_id + vf);
+	req.max_vnic_id_cnt = rte_cpu_to_le_32(bp->pf.total_vnics);
```

Comment:
So we're assuming the size of the passed-in buffer?  Maybe we should have a separate parameter to define the size.